framework/src/org/checkerframework/framework/type/AnnotationClassLoader.java

```diff
@@ -683,40 +652,53 @@ public int compare(File o1, File o2) {
      * annotation that is supported by a checker.
      *
      * @param fullyQualifiedClassName the fully qualified name of the class
-     * @return the loaded annotation class if it is defined with ElementType.TYPE_USE and is a
-     *     supported annotation, null otherwise
+     * @param issueWarnings set to true to issue a warning when a loaded annotation is not a type
+     *     annotation. It is useful to set this to true if a given annotation must be a well-defined
+     *     type annotation (eg annotation class names given as command line arguments).
+     * @return the loaded annotation class if it has a {@code @Target} meta-annotation with the
+     *     required ElementType values, and is a supported annotation by a checker. If the
+     *     annotation is not supported by a checker, null is returned.
      */
-    private final /*@Nullable*/ Class<? extends Annotation> loadAnnotationClass(
-            final String fullyQualifiedClassName) {
-        Class<?> cls = null;
+    protected final /*@Nullable*/ Class<? extends Annotation> loadAnnotationClass(
+            final String fullyQualifiedClassName, boolean issueWarnings) {
 
         try {
             // load the class
-            cls = Class.forName(fullyQualifiedClassName, true, getAppClassLoader());
-        } catch (ClassNotFoundException e) {
-            // do nothing: projects can have annotation class files and regular
-            // source files located within the same directory, and as such when
-            // it tires to load an uncompiled source file, it will throw
-            // ClassNotFoundException
-        }
-
-        // ensure that the freshly loaded class is an annotation, and has
-        // the @Target annotation
-        if (cls != null && cls.isAnnotation() && cls.getAnnotation(Target.class) != null) {
-            // retrieve the set of ElementTypes in the @Target
-            // meta-annotation and check to see if this annotation is
-            // supported for automatic loading
-            if (AnnotatedTypes.hasTypeQualifierElementTypes(
-                    cls.getAnnotation(Target.class).value(), cls)) {
-                // if it is supported, then subclass it as an Annotation
-                // class
-                Class<? extends Annotation> annoClass = cls.asSubclass(Annotation.class);
-
-                // see if the annotation is supported by a checker
-                if (isSupportedAnnotationClass(annoClass)) {
-                    return annoClass;
+            ClassLoader classLoader = getClassLoader();
+            if (classLoader != null) {
+                Class<?> cls = Class.forName(fullyQualifiedClassName, true, classLoader);
+
+                // see if the freshly loaded class is an annotation
+                if (cls != null && cls.isAnnotation()) {
```

Comment:
Same here. Invert the condition and directly return null.

Reply:
Fixed

Reply:
GitHub review tip: There is no need to answer "fixed".  That just clutters the conversation.
When you push a commit that changes this line, Werner's comment will be automatically closed.
Answer "fixed" only if the change is to some other line, so that GitHub will not automatically close the comment.
Thanks for the changes!